qiskit_experiments/framework/experiment_data.py

```diff
@@ -1599,6 +1545,21 @@ def analysis_results(
             )
         self._retrieve_analysis_results(refresh=refresh)
 
+        if index == 0:
```

Comment:
I think you want to check that index 0 holds fit parameters and otherwise fall back to the other branch here.

Reply:
Good catch. Added a check that the name starts with `@` since importing `PARAMS_ENTRY_PREFIX` would be circular and we already use starting with `@` as a filtering criteria when sending data to the plotter.